state/store.go

```diff
@@ -182,22 +182,6 @@ func LoadValidators(db dbm.DB, height int64) (*types.ValidatorSet, error) {
 	if valInfo == nil {
 		return nil, ErrNoValSetForHeight{height}
 	}
-
-	if valInfo.ValidatorSet == nil {
-		valInfo2 := loadValidatorsInfo(db, valInfo.LastHeightChanged)
-		if valInfo2 == nil {
-			panic(
-				fmt.Sprintf(
-					"Couldn't find validators at height %d as last changed from height %d",
-					valInfo.LastHeightChanged,
-					height,
-				),
-			)
-		}
-		valInfo2.ValidatorSet.IncrementProposerPriority(int(height - valInfo.LastHeightChanged)) // mutate
-		valInfo = valInfo2
```

Comment:
I thought we'd save valInfo2 here, not save for every height! I.e. we only call save once somebody tries to access it.

Reply:
> I thought we'd save valInfo2 here, not save for every height! I.e. we only call save once somebody tries to access it.

But the replay result is not correct when there are two or more rounds at one height.

Reply:
How so? If we're using `int(height - valInfo.LastHeightChanged)` difference

Reply:
> But the replay result is not correct when there are two or more rounds at one height.

Is that because we call `IncrementProposerPriority` per round for the same height then? Can you give an example (ideally a test) which clearly shows what you mean here @guagualvcha? That would be awesome.

Reply:
Note that the per-height proposer selection is *independent* of the per-round proposer selection. So the proposer at height=1 round=1 is the same as the proposer for height=2 round=0. Ie. see https://github.com/tendermint/tendermint/issues/3044